Source/Core/Core/AchievementManager.cpp

```diff
@@ -883,7 +883,16 @@ AchievementManager::ResponseType AchievementManager::Request(
   if (http_response.has_value() && http_response->size() > 0)
   {
     const std::string response_str(http_response->begin(), http_response->end());
-    process_response(rc_response, response_str.c_str());
+    {
+      std::lock_guard lg{m_lock};
```

Comment:
This feels jank. You're forcing *every* request to lock on the `process_response` call just because of the single `rc_api_process_fetch_game_data_response` that needs it.

I do like the idea though, we just need a way to lock this only when needed. I think that's possible if we change the Request function a bit, lemme think about it...

Reply:
The more I think about it the less I like any of the solutions I can come up with. I think you're better off just not calling `Request()` in `FetchGameData()` at all and just manually doing the request in it, that way you can actually safely lock around the `m_game_data` accesses.

I also think despite the locking there's currently race conditions around the `m_game_data` init/deinit cycle but we can think about that later...

Reply:
Ughhhh, I hated the idea of doing it that way because there's so much duplication when I do that. Fiiiiiiine.